src/Apps/Order/Routes/Shipping/index.tsx

```diff
@@ -814,7 +814,6 @@ export const ShippingRoute: FC<ShippingProps> = props => {
                     setAddressNeedsVerification(false)
                     setAddressHasBeenVerified(true)
                     setAddress({ ...address, ...chosenAddress })
```

Comment:
This actually came up earlier today. Have you tried 
```
setAddress(address => ({
  ...address,
  ...chosenAddress
}))
```

Using the callback form for setting state should ensure that the previous state is updated correctly. (See [this thread](https://artsy.slack.com/archives/C2TQ4PT8R/p1690812245052079?thread_ts=1690811471.733219&cid=C2TQ4PT8R))

Reply:
This didn't work in my testing so I'm going with the proposed solution and we can revisit in the follow up ticket

Reply:
Thanks for giving it a shot 👍